src/com/amazon/ion/impl/Marker.java

```diff
@@ -0,0 +1,34 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazon.ion.impl;
+
+/**
+ * Holds the start and end indices of a slice of the buffer.
+ */
+class Marker {
+
+    /**
+     * The type ID that governs the slice.
+     */
+    IonTypeID typeId = null;
```

Comment:
How does this get used?

Reply:
It's used for storing information about the value, e.g. its type, whether it is delimited, etc.